<launch>

  <arg name="params_path" default="$(optenv PARAMS_PATH)" />
  <arg name="uav_name" default="$(optenv UAV_NAME uav)" />
  <arg name="uav_mass" default="$(optenv UAV_MASS default)" />
  <arg name="profiler" default="$(optenv PROFILER false)" />
  <arg name="world_file" default="$(find mrs_general)/config/world_simulation.yaml" />
  <arg name="motor_params_file" default="$(find mrs_controllers)/config/$(arg params_path)/motor_params.yaml" />
  <arg name="g" default="9.8" />
  <arg name="standalone" default="true" />

  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg uav_name)_nodelet_manager"/>

  <group ns="$(arg uav_name)">

    <!-- UAV manager -->
    <node pkg="nodelet" type="nodelet" name="uav_manager" args="$(arg nodelet) mrs_uav_manager/UavManager $(arg nodelet_manager)" output="screen">

      <!-- Load the default param files -->
      <rosparam file="$(find mrs_uav_manager)/config/default/uav_manager.yaml" />

      <!-- Load the particular param files -->
      <rosparam file="$(find mrs_uav_manager)/config/$(arg params_path)/uav_manager.yaml" />

      <rosparam file="$(arg world_file)" />
      <rosparam file="$(arg motor_params_file)" />

      <param name="uav_mass" value="$(arg uav_mass)" />
      <param name="enable_profiler" type="bool" value="$(arg profiler)" />
      <param name="g" value="$(arg g)" />

      <!-- Subscribers -->
      <remap from="~odometry_in" to="odometry/odom_main" />
      <remap from="~mavros_odometry_in" to="mavros_interface/converted_mavros_odom" />
      <remap from="~control_manager_diagnostics_in" to="control_manager/diagnostics" />
      <remap from="~target_attitude_in" to="mavros/setpoint_raw/target_attitude" />
      <remap from="~attitude_command_in" to="control_manager/attitude_cmd" />
      <remap from="~mavros_state_in" to="mavros/state" />
      <remap from="~max_height_in" to="odometry/max_altitude" />
      <remap from="~height_in" to="odometry/altitude" />
      <remap from="~gains_in" to="gain_manager/current_gains" />
      <remap from="~constraints_in" to="constraint_manager/current_constraints" />
      <remap from="~landoff_diagnostics_in" to="control_manager/landoff_tracker/diagnostics" />
      <remap from="~motors_in" to="control_manager/motors" />

      <!-- Publishers -->
      <remap from="~attitude_cmd_out" to="mavros/setpoint_raw/attitude" />
      <remap from="~profiler" to="profiler" />

      <!-- Services -->
      <remap from="~takeoff_in" to="~takeoff" />
      <remap from="~land_in" to="~land" />
      <remap from="~land_home_in" to="~land_home" />
      <remap from="~takeoff_out" to="control_manager/landoff_tracker/takeoff" />
      <remap from="~land_out" to="control_manager/landoff_tracker/land" />
      <remap from="~eland_out" to="control_manager/landoff_tracker/eland" />
      <remap from="~switch_tracker_out" to="control_manager/switch_tracker" />
      <remap from="~switch_controller_out" to="control_manager/switch_controller" />
      <remap from="~motors_out" to="control_manager/motors" />
      <remap from="~emergency_goto_out" to="control_manager/emergency_goto" />
      <remap from="~enable_callbacks_out" to="control_manager/enable_callbacks" />
      <remap from="~arm_out" to="control_manager/arm" />
      <remap from="~pirouette_out" to="control_manager/pirouette" />

    </node>

  </group>

</launch>
